kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/TaxReporting1099ServiceImpl.java

```diff
@@ -0,0 +1,1806 @@
+					payment.setExcludeIndicator(Boolean.FALSE);
+					payment.setDisbursementDt(paymentDetail.getPaymentGroup().getDisbursementDate());
+
+					if (paymentInfo.isCheckACHPayment()) {
+						payment.setDisbursementNbr(paymentDetail.getPaymentGroup().getDisbursementNbr());
+						payment.setPaymentGroupId(paymentDetail.getPaymentGroup().getId());
+						if (TaxHelper.isDv(paymentDetail.getFinancialDocumentTypeCode())) {
+							payment.setDocType(TaxPropertyConstants.DV_DOCUMENT_TYPE_CODE);
+						} 
+						else {
+							payment.setDocType(paymentDetail.getFinancialDocumentTypeCode());
+						}
+					} 
+					else {
+						if (TaxHelper.isPreq(paymentDetail.getFinancialDocumentTypeCode())) {
+							payment.setDocType(PaymentRequestDocument.DOCUMENT_TYPE_NON_CHECK);
+						}
+						else if (TaxHelper.isCm(paymentDetail.getFinancialDocumentTypeCode())) {
+							payment.setDocType(VendorCreditMemoDocument.DOCUMENT_TYPE_NON_CHECK);
+						} 
+						else if (TaxHelper.isDv(paymentDetail.getFinancialDocumentTypeCode())) {
+							payment.setDocType(DisbursementVoucherDocument.DOCUMENT_TYPE_DV_NON_CHECK);
+						}
+					}
+
+					payment.setDocNbr(paymentDetail.getCustPaymentDocNbr());
+					payment.setInvoiceNbr(paymentInfo.getInvoiceNumber());
+					payment.setPoNbr(paymentDetail.getPurchaseOrderNbr());
+					payment.setFinChartCode(incomeType.getChartOfAccountsCode());
+					payment.setPaymentTypeCode(incomeType.getIncomeType().getIncomeTypeBox());
+
+					// set as negative for CM
+					if (TaxHelper.isCm(paymentDetail.getFinancialDocumentTypeCode())) {
+						payment.setAcctNetAmount(incomeType.getAmount().negated());
+					} 
+					else {
+						payment.setAcctNetAmount(incomeType.getAmount());
+					}
+
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("saving payment for payee=" + payment.getPayeeId() + ", incomeType=" + incomeType.getIncomeType().getIncomeTypeBox() + ", amount=" + incomeType.getAmount());
+					}
+
+					businessObjectService.save(payment);
+					retval++;
+				} 
+				else {
+					LOG.warn("unable to find documnet payment information for payee=" + payeeId + ", incomeType=" + incomeType.getIncomeType().getIncomeTypeBox() + ", documentNumber=" + paymentDetail.getCustPaymentDocNbr());
+				}
+			}
+		}
+
+		return retval;
+	}
+
+	private int saveNonPdpDVPayments(Payee payee, String documentNumber) throws Exception {
+		int retval = 0;
+
+		DisbursementVoucherDocument doc = businessObjectService.findBySinglePrimaryKey(DisbursementVoucherDocument.class, documentNumber);
+
+		if (doc != null) {
+			List<DocumentIncomeType<DisbursementVoucherDocument>> incomeTypes = get1099DocumentIncomeTypes(KFSConstants.DOCUWARE_DV_DOC_TYPE, documentNumber);
+
+			// if we found income types then create and save payments
+			if (incomeTypes != null) {
+				for (DocumentIncomeType<DisbursementVoucherDocument> incomeType : incomeTypes) {
+					// get a new Payment that is pre-populated with PaymentDetail information
```

Comment:
It seems like this could then be abstracted or parameterized but I am fine with leaving it.